### PR TITLE
Permissions Mixins

### PR DESCRIPTION
**Pull request by svenauhagen**
Hi,

these are two permission mixins for the websocketconsumer classes to check user permissions on connect.

I am not sure how to test that, more precise how I can create a testserver to start a consumer to check for this in a test.

Best
Sven

### CODE REVIEW

File: channels/generic/mixins.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class PermissionsMixin(object):
```
Comment (auvipy) [inline channels/generic/mixins.py]: explicit inherit from object is not needed in python3

File: channels/generic/mixins.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        Check for connect permissions
+        '''
+        if self.perm:
+            if self.scope["user"] and self.scope["user"].is_authenticated and self.scope["user"].has_perms(self.perm):
+                return True
+
+        return False
+
+
+class AsyncPermissionsMixin(object):
```
Comment (auvipy) [inline channels/generic/mixins.py]: same goes here..

**Comment by svenauhagen** [comment]
Thanks got it

**Comment by andrewgodwin** [comment]
Thanks for the idea, but I'm afraid I'm not going to accept these in - first of all, I don't think mixins make for good user-facing APIs, and I'd rather these were written as a simple `check_permissions` function that raised `AcceptConnction` or `DenyConnection`.

Secondly, this is the sort of code I don't think belongs in Channels itself as it's too specific to one way of implementation, but if you like, you should start a third-party app with pluggable features like this (as you would do for Django itself).

**Comment by svenauhagen** [comment]
Got it thanks. I am missing a permission check for consumers to get a fine grain check per user for my projects. Would be great to see something like that in channels in the future.

**Comment by andrewgodwin** [comment]
We already give you a `connect()` method where you can easily do a permission check or whatever check you want to let a user in; I doubt we will go beyond that in the near future. Keeping the core code simple and maintainable requires not extending too far, sorry!

**Pull request closed without merging**